skyportal/plot.py

```diff
@@ -3092,11 +3093,18 @@ async def instrument_log_plot(
 
     """
 
-    instrument = session.scalars(
-        Instrument.select(session.user_or_token).where(Instrument.id == instrument_id)
-    ).first()
-    if instrument is None:
-        raise ValueError(f'Cannot find instrument with ID "{instrument_id}"')
+    stmt = InstrumentLog.select(session.user_or_token).where(
+        InstrumentLog.instrument_id == instrument_id
+    )
+    if start_date is not None:
+        stmt = stmt.where(InstrumentLog.end_date >= start_date)
+
+    if end_date is not None:
+        stmt = stmt.where(InstrumentLog.start_date <= end_date)
+
+    instrument_logs = session.scalars(stmt).all()
+    if len(instrument_logs) == 0:
```

Comment:
This causes frontend issues I think. The frontend should receive a dictionary with the Bokeh expected structure, but empty data. Not a list of 3 null values (when inspecting you can see a json error).

Reply:
I fixed Plot.jsx to check for ! null.